src/main/java/com.gigssandbox/Activity.java

```diff
@@ -0,0 +1,33 @@
+package com.gigssandbox;
+
+import com.gigssandbox.IO.Input;
+import com.gigssandbox.IO.Output;
+import com.gigssandbox.command.Command;
+import com.gigssandbox.command.CommandFromStringArray;
+import com.gigssandbox.command.CommandType;
+
+class Activity {
+    private Input input;
+    private Output output;
+    private MainService mainService;
+
+    Activity(Input input, Output output) {
+        this.input = input;
+        this.output = output;
+        this.mainService = new MainService();
+    }
+
+    void beginReceivingAndProcessingCommands() {
```

Comment:
start() or startActivity(). Outer classes should not know what your class does inside.